File: microservice-repair/src/main/java/com/autofix/microservicerepair/services/ReparacionService.java
package com.autofix.microservicerepair.services;

import com.autofix.microservicerepair.entities.Reparacion;
import com.autofix.microservicerepair.repositories.ReparacionRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ReparacionService {
    @Autowired
    ReparacionRepository reparacionRepository;

    public List<Reparacion> crearReparaciones(List<Reparacion> reparaciones) {
        return reparacionRepository.saveAll(reparaciones);
    }

    public List<Reparacion> obtenerReparaciones() {
        return reparacionRepository.findAll();
    }

    public List<String> obtenerTipoReparaciones() {
        return reparacionRepository.obtenerTipoReparaciones();
    }
}

File: microservice-frontend/src/services/reparacion.service.js
import axios from "axios";

const REPARACION_API_URL = `http://localhost:8080/reparacion/`;

function crearReparaciones(reparaciones) {
    return axios.post(REPARACION_API_URL, reparaciones);
}

function obtenerReparaciones() {
    return axios.get(REPARACION_API_URL);
}

export default { crearReparaciones, obtenerReparaciones };